apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/FilterChainInstrumentation.java

```diff
@@ -64,9 +66,25 @@ public FilterChainInstrumentation(ElasticApmTracer tracer) {
             .and(takesArgument(1, named("javax.servlet.ServletResponse")));
     }
 
-    @Override
-    public Class<?> getAdviceClass() {
-        return ServletApiAdvice.class;
+    @Nullable
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    private static Transaction onEnterServletService(@Advice.Origin Class<?> clazz,
+                                                     @Advice.Argument(0) ServletRequest servletRequest) throws Throwable {
+        return (Transaction) MethodHandleDispatcher
```

Comment:
Referencing agent types in advice methods might not be possible anymore once we load the agent from an isolated class loader.